Add clear errors

kenconnor opened the pull request:
Taking over https://github.com/odriverobotics/ros_odrive/pull/14
The former PR broke file structure and I made a brand new PR.

I tested it on S1. I called `/request_axis_state` with wrong state, called `/clear_errors` and then called `/request_axis_state` with correct state. I confirmed that `procedure_result` is changed to 0.

samuelsadok reviewed (commented):
> I tested it on S1. I called /request_axis_state with wrong state, called /clear_errors and then called /request_axis_state with correct state. I confirmed that procedure_result is changed to 0.

To be clear, an explicit clear_errors call should _not_ be considered a requirement for entering CLOSED_LOOP_CONTROL. On a high level, an explicit call to clear_errors should _only_ be used to make the LED not be red. See explanation [here](https://github.com/odriverobotics/ros_odrive/pull/14#discussion_r1547950910). If entering CLOSED_LOOP_CONTROL is the goal, clearing errors should be handled transparently by (future) ODrive firmware, or, for compatibiliy with older firmware, transparently in `request_state_callback()`. 

If clearing the LED color is the goal of this PR, then the approach looks mostly good. The git history is still a bit convoluted and the PR contains unrelated formatting changes. Please use git rebase / cherry-pick / amend so that the PR consists of a single commit and doesn't contain unrelated formatting changes. Please also add a note in README about the new service call.

kenconnor commented:
I think S1 needs to call clear_errors() to re-enable the brake resistor, but under what circumstances is the brake resistor disarmed? Since I use S1, I would like to enable and use the brake resistor if it is disabled.

I made the following edits:
- Removed whitespace unrelated to the PR from the commit
- Added to the README

I was concerned that the contribution of liborw in #14  might disappear, so I decided not to consolidate the commits to preserve the commit history. Should we consolidate all changes, including the additions to the README, into a single commit (add service to clear errors)?

samuelsadok commented on odrive_node/README.md:
```diff
@@ -50,6 +50,10 @@ For information about installation, prerequisites and getting started, check out
 
+* `/clear_errors`: Clears disarm_reason and procedure_result and re-arms the brake resistor if applicable
+
+  Change the axis state after clearing the errors, as the axis state is reset to idle when an error occurs.
```
This sentence I find a bit confusing. If I understand correctly, you meant something like: "If the axis dropped into IDLE because of an error, clearing the errors does not put the axis back into CLOSED_LOOP_CONTROL. To do so, you must request CLOSED_LOOP_CONTROL again explicitly."

kenconnor replied:
That is what I was trying to say. I am sorry that my expression is not clear and thank you for your correction.

samuelsadok reviewed (commented):
Yes that's true, if you need the brake resistor in your system to be running even if the motor is off (e.g. to sink power from another ODrive), then that's another reason to call clear_errors(). The brake resistor disarms on conditions that specifically restrict the brake resistor's ability to run safely: overvoltage, undervoltage, configuration error and low level system error. We're considering to change this in future firmware versions to re-enable it automatically when the conditions allow it.

Formatting looks good now.

If you wish to preserve authorship, there are multiple ways to do it. E.g. you can still squash the original commits into one, and put additional changes in a separate one, or mention [co-authorship in the commit message](https://stackoverflow.com/a/7442255).

kenconnor commented:
@samuelsadok 
I have made the following changes:
- Squashed original commits
- Left Libor as a co-author
- Changed the sentence in README

samuelsadok commented:
looks good, thanks!
